src/convertToObject.js

```diff
@@ -5,8 +5,24 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const cssObject = {};
+
+  const rules = sourceString
+    .split(/;(?![^(]*\))/)
+    .filter((rule) => rule.trim() !== '');
+
+  rules.forEach((rule) => {
+    const [property, ...valueParts] = rule.split(':');
+    const value = valueParts.join(':').trim();
+
+    if (property && value) {
+      cssObject[property.trim()] = value.replace(/;\s*$/, '').trim();
```

Comment:
The `.replace(/;\s*$/, '').trim()` seems unnecessary because at this point in the code, the semicolon should already have been removed by the `.split()` operation on line 13. If this is meant to handle a specific edge case, please document it clearly.